### PR TITLE
chore: init templates use node jest environment

### PR DESCRIPTION
**Pull request by MrArnoldPalmer**
Remove usage of the `jsdom` test environment in init templates to speed
up unit testing by default.

Testing: ran cdk init --language=(typescript|javascript) against local build of
CLI then ran yarn test to verify that the testing config was valid and jest correctly
used the node environment.

fix: #14630

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

File: packages/aws-cdk/lib/init-templates/v1/app/javascript/jest.config.js [hunk 1]
```diff
@@ -0,0 +1,3 @@
+module.exports = {
```
Comment (nija-at) [inline packages/aws-cdk/lib/init-templates/v1/app/javascript/jest.config.js:1]: Can you add a comment to this PR titled 'Testing' on how you tested this change? At a minimum, that the experience doesn't regress.

context: we don't have very good automated test coverage for our init templates, and have got it wrong many times.

**Review by nija-at — CHANGES_REQUESTED** [review]
Dang! Should we be setting this to all of the CDK repo jest set up too? 

What kind of improvements do you see/expect from this?

**Comment by MrArnoldPalmer** [comment]
> Dang! Should we be setting this to all of the CDK repo jest set up too?
> 
> What kind of improvements do you see/expect from this?

This is already correctly configured in the cdk jest environment via our [root jest config](https://github.com/aws/aws-cdk/blob/master/tools/cdk-build-tools/config/jest.config.js#L8), so no gains for us 😢

**Comment by MrArnoldPalmer** [comment]
Testing: ran `cdk init --language=(typescript|javascript)` against local build of CLI then ran `yarn test` to verify that the testing config was valid and jest correctly used the `node` environment.

**Review by nija-at — APPROVED** [review]

**Pull request merged**